int32 goal_number
---
int32 final_final
---
int32 current_number
